packages/jellyfish-api-core/src/category/blockchain.ts

```diff
@@ -164,6 +164,36 @@ export class Blockchain {
     return await this.client.call('getrawmempool', [verbose], 'bignumber')
   }
 
+  /**
+   * Get all in-mempool ancestors for a given transaction as string[]
+   *
+   * @param {string} txId the transaction id
+   * @param {boolean} verbose false
+   * @return {Promise<string[]>}
+   */
+  getMempoolAncestors (txId: string, verbose?: false): Promise<string[]>
+
+  /**
+   * Get all in-mempool ancestors for a given transaction as json object
+   *
+   * @param {string} txId the transaction id
+   * @param {boolean} verbose true
+   * @return {Promise<MempoolTx>}
+   */
+  getMempoolAncestors (txId: string, verbose?: true): Promise<MempoolTx>
+
+  /**
+   * Get all in-mempool ancestors if txId is in mempool as string[] if verbose is false
+   * else as json object
+   *
+   * @param {string} txId the transaction id
+   * @param {boolean} verbose default = false, true for json object, false for array of transaction ids
+   * @return {Promise<string[] | MempoolTx>}
+   */
+  async getMempoolAncestors (txId: string, verbose?: boolean): Promise<string[] | MempoolTx> {
+    return await this.client.call('getmempoolancestors', [txId, verbose], 'bignumber')
```

Comment:
A little history about this design, back when we designed the JSON-RPC interfaces, we wanted to make the object field DX-friendly. Hence, the invention allowed a declaration to declare where the BigNumber or numbers are. The idea was to be "ACCOMMODATING", and make them a number when the value isn't sensitive to precision, lost such as "maxFees" or height.

Although the intention was sound, we have some regrets now as it is unnecessarily complicated to declare new fields. Almost impossible when dealing with unknown key-value objects (this RPC, `MempoolTx`). Maybe it is better just to make them all BigNumber and call it a day.

- https://github.com/JellyfishSDK/jellyfish/issues/18
- https://github.com/JellyfishSDK/jellyfish/pull/97

Reply:
left the precision as 'bignumber' (same as `getMempoolEntry`) for now

Reply:
Yeah, we have to for KeyVault objects.